pkg/models/model.go

```diff
@@ -141,80 +129,28 @@ func (m *Model) setNodeMap() error {
 	return nil
 }
 
-func getNetworkLayers(name string, swmap map[string]string) ([]string, error) {
-	sw := make(map[string]bool)
-	res := []string{}
+func getNetworkLayers(sw *Switch, swmap map[string]*Switch) ([]string, map[string]string, error) {
+	visited := make(map[string]bool)
+	layers := []string{}
+	metadata := make(map[string]string)
+
 	for {
+		name := sw.Name
 		// check for circular switch topology
-		if _, ok := sw[name]; ok {
-			return nil, fmt.Errorf("circular topology for switch %q", name)
+		if _, ok := visited[name]; ok {
+			return nil, nil, fmt.Errorf("circular topology for switch %q", name)
```

Comment:
Nit: A little more easy to understand error message.
"circular dependency detected in topology for switch"